12/README.md

```diff
@@ -0,0 +1,165 @@
+
+*Tango Server* (server) - a process that runs on a machine.
+
+*Tango Device* (device) - a thread within a *Tango Server* that implements *Tango Class* i.e. has *Tango Commands, Attributes, Pipes, Properties*  etc.
+
+*Tango Admin Device* (admin) - a pre-defined *Tango Device* i.e. a *Tango Device* with fixed number *Tango Commands, Attributes, Pipes* [RFC-3, RFC-4,RFC-?]  etc.
+
+For more details see [RFC-2](https://github.com/tango-controls/rfc/tree/draft/2)
+
+*Tango Client* (client) - a process that instantiates communication with *Tango Server* using Request-Reply protocol [RFC-10].
+
+*Tango Client Library* (client kernel library) - library that encapsulates communication with *Tango Server* and provides *Tango Client API* to a user.
+
+*Tango Client API* (client API) - an API that user uses to develop applications based on communication with *Tango Server*s. 
+
+*Tango Server Library* (server kernel library) - library that encapsulates *Tango Server* features and provides *Tango Server API* to a user.
+
+*Tango Server API* (server API) - an API that user uses to develop *Tango Device* functionality.
+
+*Upstream Tango Server/Device* (upstream server) - a *Tango Device* to which updates *client* subscribes for.  
+
+*Polling thread* (polling thread) - a thread within *Tango Server* that polls periodically defined *Tango Device*'s attributes/sometimes commands.
+
+*Tango Publish-Subscriber protocol* (pub-sub) - is a sub-set of *Tango Client-Server protocol* [RFC-?] that defines a relationship between *client* and *server* established via a negotiation phase and lasts till explicitly broken. During this relationship client receives updates published by *upstream server*. Updates are sourced from server's *polling thread* or directly through *server API*. Updates are filtered according to *device*'s configuration.
+
+*Tango Event* (event) - an update sent by *server*.
+
+*Tango Event Callback* (callback) - a piece of code that *client library* executes when an update is received.
+
+## Definitions
+
+This section offers a number of ABNF formal definitions for entities repeatedly used below in this document. 
+
+```ABNF
+EVENT_TYPE = "QUALITY_EVENT" | "INTERFACE_CHANGE" | "PIPE" | "ATTR_CONF_EVENT" |  "CHANGE_EVENT" | "PERIODIC_EVENT" | "ARCHIVE_EVENT" | "USER_EVENT" | "HEARTBEAT"
+
+HEARTBEAT_ENDPOINT = "(upstream server FQDN):port"  
+
+EVENT_ENDPOINT = "(upstream server FQDN):port"
+
+HEARTBEAT_CHANNEL = "(upstream admin device FQDN)/heartbeat" 
+
+EVENT_CHANNEL = "(upstream admin device FQDN)/(attr).EVENT_TYPE[_idl5]" 
+```
+
+## Runtime requirements
+
+Client and server are up and running. Server is reachable from client i.e. may communicate using Request-Reply protocol [RFC-10]. 
+
+## Publisher-Subscriber protocol
+
+![Subscription_1.png](images/Subscription_1.png)
+
+The main goal of the this protocol implementation is to allow client receive events from server:
```

Comment:
The main goal of the this protocol implementation is to allow a Client to receive events from a Server: